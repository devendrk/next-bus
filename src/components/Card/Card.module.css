.card {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  box-shadow: 0 4px 8px 5px rgba(0, 0, 0, 0.205);
  background-color: rgba(255, 255, 255, 0.884);
  width: 500px;
  overflow: scroll;
  max-height: 95%;
}
.cardHeader {
  text-align: center;
  padding: 32px 0;
  font-size: 24px;
  font-weight: 700;
  margin: 0;
  background: skyblue;
}
.stationNumber {
  font-weight: 200;
  color: #7a7a7a;
}
.cardSubHeader {
  display: flex;
  justify-content: space-between;
  padding: 0 10%;
  font-weight: 700;
  color: #7a7a7a;
}

@media screen and (max-width: 1100px) {
  .card {
    width: 500px;
  }
}

@media screen and (max-width: 992px) {
  .card {
    position: relative;
    width: 100vw;
    margin: 0 0;
  }
}
