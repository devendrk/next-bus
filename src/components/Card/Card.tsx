import React from "react";
import styles from "./Card.module.css";

interface NextBusses {
  scheduledArrival: number;
  headsign: number;
}
interface CardProps {
  /** Name of the stop */
  stopName?: string;
  /** Bus stop code  */
  stopNumber?: string;
  /** List of  all busses that will arrives to this Bus stop */
  nextBuses: NextBusses[];
}

const Card: React.FC<CardProps> = ({ nextBuses, stopNumber, stopName }) => {
  /**
   * @param {number} arrivalTime  time in seconds from last 00:00 ( 12:00 am)
   * @returns {string}
   */
  const getArrivalTime = (arrivalTime: number): string => {
    var date = new Date(0);
    date.setSeconds(arrivalTime);
    var timeString = date.toISOString().substr(11, 5);
    return timeString;
  };
  const renderSchedule = () => {
    return nextBuses.map((nb: any) => (
      <div
        className={styles.cardBody}
        key={nb.headsign.length * nb.scheduledArrival}
      >
        <p>{nb.headsign}</p>
        <p>{getArrivalTime(nb.scheduledArrival)}</p>
      </div>
    ));
  };

  return (
    <div className={styles.card}>
      <h1 className={styles.cardHeader}>
        {stopName} <span className={styles.stopNumber}>( {stopNumber} )</span>
      </h1>
      <div className={styles.cardSubHeader}>
        <p>Heading to</p>
        <p>Arrives at </p>
      </div>
      {renderSchedule()}
    </div>
  );
};

export default Card;
