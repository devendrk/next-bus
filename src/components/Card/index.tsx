import React from "react";
import BusSchedule from "../BusSchedule";
import styles from "./Card.module.css";

import { Stops } from "../../types/index";

interface CardProps {
  /** Name of the stop */
  stopName: string;
  /** Bus stop code  */
  stopNumber: string;
  /** List of  all busses that will arrives to this Bus stop */
  nextBuses: Stops[];
}

const Card: React.FC<CardProps> = ({ nextBuses, stopNumber, stopName }) => {
  return (
    <div className={styles.card}>
      <h1 className={styles.cardHeader}>
        {stopName} <span className={styles.stopNumber}>( {stopNumber} )</span>
      </h1>
      <div className={styles.cardSubHeader}>
        <p>Line</p>
        <p>Heading to</p>
        <p>Arrives at </p>
      </div>
      <BusSchedule nextBuses={nextBuses} />
    </div>
  );
};

export default Card;
