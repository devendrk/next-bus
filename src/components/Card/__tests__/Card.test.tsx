import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import Card from "..";

describe("<Card />", () => {
  const stopName = "stopName";
  const stopNumber = "V1056";
  const nextBuses = [
    {
      headsign: "Helsinki",
      scheduledArrival: 4200,
      trip: { route: { shortName: "abc", id: "abc" } },
    },
  ];
  it("Card", () => {
    const { container } = render(
      <Card stopName={stopName} stopNumber={stopNumber} nextBuses={nextBuses} />
    );
    expect(container).not.toBeNull();
  });
});
