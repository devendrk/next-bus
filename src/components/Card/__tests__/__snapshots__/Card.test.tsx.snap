// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Card 1`] = `
<div>
  <div
    class="card"
  >
    <h1
      class="cardHeader"
    >
      stopName
       
      <span
        class="stopNumber"
      >
        ( 
        V1056
         )
      </span>
    </h1>
    <div
      class="cardSubHeader"
    >
      <p>
        Line
      </p>
      <p>
        Heading to
      </p>
      <p>
        Arrives at 
      </p>
    </div>
    <div>
      <div
        class="cardBody"
      >
        <p>
          abc
        </p>
        <p
          class="short-name"
        >
          Helsinki
        </p>
        <p>
          01:10
        </p>
      </div>
    </div>
  </div>
</div>
`;
